fgpyo/util/metric.py

```diff
@@ -334,3 +344,330 @@ def fast_concat(*inputs: Path, output: Path) -> None:
             io.write_lines(
                 path=output, lines_to_write=list(io.read_lines(input_path))[1:], append=True
             )
+
+
+def is_metric(cls: Any) -> bool:
+    """True if the given class is a Metric."""
+
+    return (
+        isclass(cls)
+        and issubclass(cls, Metric)
+        and (dataclasses.is_dataclass(cls) or attr.has(cls))
+    )
+
+
+@dataclass(kw_only=True)
+class MetricFileFormat:
+    """
+    Parameters describing the format and configuration of a delimited Metric file.
+
+    Most of these parameters, if specified, are passed through to `csv.DictReader`/`csv.DictWriter`.
+    """
+
+    delimiter: str = "\t"
+    comment: str = "#"
```

Comment:
Should we anticipate multiple comment prefixes?

```suggestion
    comment_prefixes: set[str] = {"#"}
```

Reply:
👍

Reply:
I think this is unlikely to occur in practice and makes the interface clunkier.

Could you provide an example of a file format that would have multiple prefixes? I can't think of one off the top of my head.

Reply:
UCSC BED files come to mind, but they also lack a header with named fields entirely. This is one of those API decisions that's hard to go back on since it will be a breaking change in the future, but not now.

Reply:
If I'm understanding the API correctly, this wouldn't only activate for mixed comment prefixes in the same file, but allow for setting multiple comment prefixes across files too?